labs-ios-starter/Map/MapViewController.swift

```diff
@@ -7,24 +7,43 @@
 //
 
 import UIKit
+import MapKit
 
 class MapViewController: UIViewController {
 
+    @IBOutlet weak var mapView: MKMapView!
+    
+    let locationManager = CLLocationManager()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        locationManager.requestWhenInUseAuthorization()
+        
+        if CLLocationManager.locationServicesEnabled() {
+            locationManager.delegate = self
+            locationManager.desiredAccuracy = kCLLocationAccuracyBest
+            locationManager.startUpdatingLocation()
+        }
+    }
+}
 
-        // Do any additional setup after loading the view.
+extension MapViewController : CLLocationManagerDelegate {
+    private func locationManager(manager: CLLocationManager, didChangeAuthorizationStatus status: CLAuthorizationStatus) {
+        if status == .authorizedWhenInUse {
+            locationManager.requestLocation()
+        }
     }
     
```

Comment:
extensions should be at the bottom of the swift file. Just good code formatting practices

Reply:
that is the bottom of the file

Reply:
it didn't look like it. Not calling you a liar but was showing that was line 35 and there was code under it going to line 48 or something like that. My bad